src/components/user.js

```diff
@@ -32,6 +48,37 @@ export default class User extends React.Component {
         this.handleClickSubmit  = this.handleClickSubmit.bind(this);
     }
 
+    _validateField (fieldName, value) {
+        let { formErrors, valid } = this.state;
+        let ans;
+
+        switch (fieldName) {
+            case 'name':
+                valid.name      = value !== undefined && value.length > 0;
+                formErrors.name = valid.name ? '' : 'Name can not be empty.';
+                ans             = valid.name;
+                break;
+            case 'phone':
+                valid.phone      = value !== undefined && value.match(/^([\d.\-+x()]+)$/i);
+                formErrors.phone = valid.phone ? '' : 'Phone is invalid.';
+                ans              = valid.phone;
+                break;
+            case 'email':
+                valid.email      = value !== undefined && !!value.match(/^([\w.-]+)@([\w-]+\.)+([\w]{2,})$/i);
+                formErrors.email = valid.email ? '' : 'Email is invalid.';
+                ans              = valid.email;
+                break;
+            default:
+                break;
+        }
+        valid.form = valid.email && valid.name && valid.phone;
```

Comment:
(minor)
I'm not sure that we need this property. 
We can use a variable in the `render` method, e.g.
```js
const isFormFieldsValid = valid.email && valid.name && valid.phone;
```